addons/mail/static/src/models/composer/composer.js

```diff
@@ -602,7 +602,12 @@ function factory(dependencies) {
         _getCommandFromText(content) {
             if (content.startsWith('/')) {
                 const firstWord = content.substring(1).split(/\s/)[0];
-                return this.env.messaging.commands.find(command => command.name === firstWord);
+                return this.env.messaging.commands.find(command => {
+                    if (command.channel_types) {
+                        return command.name === firstWord && command.channel_types.includes(this.thread.channel_type);
+                    }
+                    return command.name === firstWord
```

Comment:
@seb-odoo as discussed on PR 60165, I added the filter condition based on the channel types

Reply:
Make it the same structure as the other filter, in pseudo code:
if name != firstword return false
if channel type return channel type includes
(else implicit) return true